packages/twenty-front/src/modules/views/hooks/internal/usePersistViewGroupRecords.ts

```diff
@@ -0,0 +1,116 @@
+import { useCallback } from 'react';
+import { useApolloClient } from '@apollo/client';
+import { v4 } from 'uuid';
+
+import { triggerCreateRecordsOptimisticEffect } from '@/apollo/optimistic-effect/utils/triggerCreateRecordsOptimisticEffect';
+import { useObjectMetadataItem } from '@/object-metadata/hooks/useObjectMetadataItem';
+import { useObjectMetadataItems } from '@/object-metadata/hooks/useObjectMetadataItems';
+import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+import { useCreateOneRecordMutation } from '@/object-record/hooks/useCreateOneRecordMutation';
+import { useUpdateOneRecordMutation } from '@/object-record/hooks/useUpdateOneRecordMutation';
+import { GraphQLView } from '@/views/types/GraphQLView';
+import { ViewGroup } from '@/views/types/ViewGroup';
+
+export const usePersistViewGroupRecords = () => {
+  const { objectMetadataItem } = useObjectMetadataItem({
+    objectNameSingular: CoreObjectNameSingular.ViewGroup,
+  });
+
+  const { createOneRecordMutation } = useCreateOneRecordMutation({
+    objectNameSingular: CoreObjectNameSingular.ViewGroup,
+  });
+
+  const { updateOneRecordMutation } = useUpdateOneRecordMutation({
+    objectNameSingular: CoreObjectNameSingular.ViewGroup,
+  });
+
+  const { objectMetadataItems } = useObjectMetadataItems();
+
+  const apolloClient = useApolloClient();
+
+  const createViewGroupRecords = useCallback(
+    (viewGroupsToCreate: ViewGroup[], view: GraphQLView) => {
+      if (!viewGroupsToCreate.length) return;
```

Comment:
Shouldn't we use a createMany ?

Reply:
I was thinking the same, but I think it was due to a backend limitation because of the response of the server that was not including the updated of created objects.
But after taking another look it seems it should be fine.
We can maybe take a look on that later ? As it's done the same way for ViewFilter, ViewField, etc...
cc @charlesBochet

Reply:
Sure ! I created an issue https://github.com/twentyhq/twenty/issues/7989